### ISSUE TITLE
convenience_type false positive on Function Builder attributed types

### ISSUE THREAD

**Issue by theoriginalbit** [realm/SwiftLint#3144]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

Similar to #3033 except specifically `convenience_type` is producing a warning on types that have the Swift 5.1 Function Builder (`@_functionBuilder`).

Perfect world perhaps the rule have a configuration that allows specifying attributes that disable the rule – maybe with defaults of `@objc` and `@_functionBuilder`.

### Environment

* SwiftLint version (run swiftlint version to be sure)?
0.39.1
* Installation method used (Homebrew, CocoaPods, building from source, etc)?
brew
* Do you have a sample that shows the issue?

```swift
public protocol AnySetter {}

// This triggers a violation: convenience_type
@_functionBuilder
public struct AnySetterBuilder {
    static func buildBlock(_ setters: AnySetter...) -> [AnySetter] {
        return setters
    }
}
```

**Comment by marcelofabri** [comment]
Can a `@_functionBuilder` be an `enum`? This is what the rule is suggesting. If not, we should add an exception to it. 

But if it can, I'm not sure if we should treat this any different than a regular `struct`.

**Comment by theoriginalbit** [comment]
You know what. I think you might be right. Most resources out there will use `struct`, but I think it should be fine to be an enum because it requires only static functions.
